c10/cuda/CUDADeviceAssertionHost.cpp

```diff
@@ -0,0 +1,366 @@
+#include <c10/cuda/CUDADeviceAssertionHost.h>
+#include <c10/util/Backtrace.h>
+#include <c10/util/Exception.h>
+#include <c10/util/irange.h>
+#include <cuda_runtime.h>
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <thread>
+
+#define CHECK_CUDA_API_CALL_WITHOUT_CHECKING_DEVICE_ASSERTS \
```

Comment:
Nit: can you please make it look like a function?
```suggestion
#define CHECK_CUDA_API_CALL_WITHOUT_CHECKING_DEVICE_ASSERTS() \
```

Reply:
Done, thanks.